src/internal_events/decoder.rs

```diff
@@ -26,25 +28,65 @@ impl<'a> InternalEvent for DecoderDeserializeFailed<'a> {
 }
 
 #[derive(Debug)]
-pub struct EncoderFramingFailed<'a> {
+pub struct EncoderFramingError<'a> {
     pub error: &'a codecs::encoding::BoxedFramingError,
 }
 
-impl<'a> InternalEvent for EncoderFramingFailed<'a> {
+impl<'a> InternalEvent for EncoderFramingError<'a> {
     fn emit(self) {
         warn!(message = "Failed framing bytes.", error = %self.error, internal_log_rate_secs = 10);
+        error!(
+            message = "Events dropped.",
+            count = 1,
+            error = %self.error,
+            error_type = error_type::ENCODER_FAILED,
+            stage = error_stage::SENDING,
```

Comment:
Do we have a written guide somewhere that helps with classifying `SENDING` vs `PROCESSING`? To me, this would rather fall into the "processing" category instead of "sending"

Reply:
I'm not sure we do have that, I would say `PROCESSING` cover everything that modifies the bytes buffer (incl. encoding and compression) and that `SENDING` would cover writing the buffer to the destination. But I don't think we have a written statement about `PROCESSING` vs. `SENDING` boundaries.

Reply:
@prognant do we need a follow-up for this?

Reply:
I considered these stages broadly aligned with the component "type" which likely isn't totally accurate, but it is a gray area for sure.

I picked `SENDING` here as we encoded the event to send it 🤷 - rather than us doing internal processing and transformations.